sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/ForbidCertainMethodsCheck.java

```diff
@@ -0,0 +1,281 @@
+     */
+    public void setArgumentCount(String argumentCount) throws CheckstyleException {
+        this.argumentCount = argumentCount;
+        if (argumentCount.trim().length() > 0) {
+            final String[] rangeTokens = COMMA_REGEX.split(argumentCount);
+            for (String oneToken : rangeTokens) {
+                this.argumentCountRanges.add(new IntRange(oneToken));
+            }
+        }
+    }
+
+    /**
+     * Return null if the regex of the pattern is empty string, otherwise return the pattern.
+     * @param pattern to be checked
+     * @return null or supplied pattern
+     */
+    private static Pattern getPatternOrNull(Pattern pattern) {
+        final Pattern returnValue;
+        if (pattern.pattern().equals("")) {
+            returnValue = null;
+        }
+        else {
+            returnValue = pattern;
+        }
+        return returnValue;
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {
+            TokenTypes.METHOD_CALL,
+        };
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        switch (ast.getType()) {
+            case TokenTypes.METHOD_CALL:
+                final DetailAST dot = ast.getFirstChild();
+                // method that looks like: method()
+                final String methodNameInCode;
+                if (dot.getType() == TokenTypes.IDENT) {
+                    methodNameInCode = dot.getText();
+                }
+                // method that looks like: obj.method()
+                else {
+                    methodNameInCode = dot.getLastChild().getText();
+                }
+                final int numArgsInCode = dot.getNextSibling().getChildCount(TokenTypes.EXPR);
+                checkForbiddenMethod(ast, methodNameInCode, numArgsInCode);
+                break;
+            default:
+                Utils.reportInvalidToken(ast.getType());
+                break;
+        }
+    }
+
+    /**
+     * Check if the method/constructor call against defined rules.
+     * @param ast ast for the call
+     * @param name ruleName of the the method
+     * @param argCount number of arguments of the method
+     */
+    private void checkForbiddenMethod(DetailAST ast, String name, int argCount) {
+        if (methodName != null) {
+            final boolean methodNameMatches = methodName.matcher(name).matches();
+            boolean argCountMatches = argumentCountRanges.isEmpty();
+            for (IntRange intRange : argumentCountRanges) {
+                if (intRange.contains(argCount)) {
+                    argCountMatches = true;
+                }
+            }
+            if (methodNameMatches && argCountMatches) {
+                if (argumentCountRanges.isEmpty()) {
```

Comment:
Move this `if` statement to where `argCountMatches` is defined.
`else` will be rest of code below moved to line starting with `boolean argCountMatches = false;`.

Reply:
Done.